drivers/storage/portworx/component/plugin.go

```diff
@@ -0,0 +1,359 @@
+	err := k8s.ParseObjectFromFile(BaseDir+filename, p.scheme, service)
+	if err != nil {
+		return err
+	}
+	service.Namespace = storageNs
+	service.OwnerReferences = []metav1.OwnerReference{*ownerRef}
+	return k8s.CreateOrUpdateService(p.client, service, ownerRef)
+}
+
+func (p *plugin) createConfigmap(filename, storageNs string, ownerRef *metav1.OwnerReference) error {
+	cm := &v1.ConfigMap{}
+	err := k8s.ParseObjectFromFile(BaseDir+filename, p.scheme, cm)
+	if err != nil {
+		return err
+	}
+	cm.Namespace = storageNs
+	cm.OwnerReferences = []metav1.OwnerReference{*ownerRef}
+	updateDataIfNginxConfigMap(cm, storageNs)
+
+	_, err = k8s.CreateOrUpdateConfigMap(p.client, cm, ownerRef)
+	return err
+}
+
+func (p *plugin) createConsolePlugin(filename, storageNs string, ownerRef *metav1.OwnerReference) error {
+
+	if err := console.AddToScheme(p.scheme); err != nil {
+		return err
+	}
+
+	cp := &console.ConsolePlugin{}
+	if err := k8s.ParseObjectFromFile(BaseDir+filename, p.scheme, cp); err != nil {
+		return err
+	}
+
+	cp.Namespace = storageNs
+	cp.OwnerReferences = []metav1.OwnerReference{*ownerRef}
+	if cp.Spec.Backend.Service != nil {
+		cp.Spec.Backend.Service.Namespace = storageNs
+	}
+
+	if len(cp.Spec.Proxy) > 0 && cp.Spec.Proxy[0].Endpoint.Service != nil {
+		cp.Spec.Proxy[0].Endpoint.Service.Namespace = storageNs
+	}
+
+	existingPlugin := &console.ConsolePlugin{}
+	err := p.client.Get(
+		context.TODO(),
+		types.NamespacedName{
+			Name:      cp.Name,
+			Namespace: cp.Namespace,
+		},
+		existingPlugin,
+	)
+
+	if errors.IsNotFound(err) {
+		logrus.Infof("Creating %s Consoleplugin", cp.Name)
+		return p.client.Create(context.TODO(), cp)
+	} else if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (p *plugin) deleteConsolePlugin(name, storageNs string, owners ...metav1.OwnerReference) error {
+	if err := console.AddToScheme(p.scheme); err != nil {
+		return err
+	}
+
+	resource := types.NamespacedName{
+		Name:      name,
+		Namespace: storageNs,
+	}
+
+	consolePlugin := &console.ConsolePlugin{}
+	err := p.client.Get(context.TODO(), resource, consolePlugin)
+	if errors.IsNotFound(err) {
+		return nil
+	} else if err != nil {
+		return err
+	}
+
+	newOwners := k8s.RemoveOwners(consolePlugin.OwnerReferences, owners)
+
+	// Do not delete the object if it does not have the owner that was passed;
+	// even if the object has no owner
+	if (len(consolePlugin.OwnerReferences) == 0 && len(owners) > 0) ||
+		(len(consolePlugin.OwnerReferences) > 0 && len(consolePlugin.OwnerReferences) == len(newOwners)) {
+		logrus.Infof("Cannot delete consolePlugin %s/%s as it is not owned",
+			storageNs, name)
+		return nil
+	}
+
+	if len(newOwners) == 0 {
+		logrus.Infof("Deleting %s/%s consolePlugin", storageNs, name)
+		return p.client.Delete(context.TODO(), consolePlugin)
+	}
+	consolePlugin.OwnerReferences = newOwners
+	logrus.Infof("Disowning %s/%s consolePlugin", storageNs, name)
+	return p.client.Update(context.TODO(), consolePlugin)
```

Comment:
Just like other k8s objects, can you move this to the k8s util instead of keeping it here?